bazel/packaging/BUILD

```diff
@@ -0,0 +1,34 @@
+load("@rules_pkg//pkg:mappings.bzl", "pkg_files", "strip_prefix")
+load("@rules_pkg//pkg:tar.bzl", "pkg_tar")
+
+pkg_files(
+    name = "redpanda_binary_with_runfiles",
+    srcs = [
+        "//src/v/redpanda",
+    ],
+    # Including runfiles has the effect of copying all of the shared libraries needed to run the binary.
+    # Although they always retain the directory structure that is found in bazel build output.
+    include_runfiles = True,
+)
+
+pkg_files(
+    name = "redpanda_binary_flattened",
+    srcs = [
+        ":redpanda_binary_with_runfiles",
+    ],
+    prefix = "opt/redpanda/libexec",
+    # This flattens the files so that everything is directly in the `libexec` folder.
+    strip_prefix = strip_prefix.files_only(),
+)
+
+# TODO(bazel): Pull in all the system libraries that are still depended on and not packaged yet.
+# TODO(bazel): Add our wrapper redpanda script that adds LD_LIBRARY_PATH.
+pkg_tar(
```

Comment:
Currently this also includes 2 shared libraries - carried over from hwloc (because zlib and libpciaccess are bazel builds and both shared and static libraries are copied over. We can probably disable the shared libraries there somehow to fix, or just filter them out here.)

Reply:
Okay libpciaccess was a pretty easy fix, but zlib is harder because it's used by multiple other non-bazel native builds. I'll leave it for now.